ibis/tests/all/test_window.py

```diff
@@ -15,6 +17,68 @@
 )
 
 
+def ntile(x: pandas.core.groupby.generic.SeriesGroupBy, bucket: int):
+    """
+    NTILE divides given data set into a number of buckets.
+
+    It divides an ordered and grouped data set into a number of buckets
+    and assigns the appropriate bucket number to each row.
+
+    Return an integer ranging from 0 to `bucket - 1`, dividing the
+    partition as equally as possible.
+
+    Parameters
+    ----------
+    x : pandas.core.groupby.generic.SeriesGroupBy ||
+        pandas.core.groupby.generic.DataFrameGroupBy ||
+        pandas.Series || pandas.DataFrame
+    bucket: int
+
+    Returns
+    -------
+    pandas.Series
+
+    Examples
+    --------
+    >>> result = ntile(pandas.Series([12000, 9000, 8200, 7800, 7700, 6900]), 4)
+    >>> assert numpy.all(result.values == [0, 0, 1, 1, 2, 3])
+    """
+    # internal ntile function
+    def _ntile(x: pandas.Series, bucket: int):
+        n = x.shape[0]
```

Comment:
why are you actually trying to execute operations inside api definitions?

this is an anti pattern in ibis

further these are actually definitions of pandas ops and not omnisci db native ops

so am puzzled what you are attempting to accomplish 

the way to do this is to define these as pandas ops
the api then should return these ops and ibis will dispatch to them

Reply:
I just created that because AFAIK currently there is no way to test ibis ntile operation ... 
so that would be just a temporary way to test ibis ntile operations
just for clarification, by `pandas ops` do you mean 1) ibis pandas ops or really 2) pandas ops?

if it is 2) .. as it could take an extra time, probably I would prefer to remove ntile here and implement that in a different PR.

Reply:
I am not sure you are following. This completely breaks the ibis style & philosophy. I am really not sure what your end goal is Does omniscidb actually have these functions? if so certainly you can add them that backend, but this shouldn't have anything to do with pandas.

Reply:
yes .. maybe I am not following you ...  what do you mean by ` api definitions`? because I added that just to `test_window.py` it shouldn't be available into the API.

I can remove this code with no problem .. my question is .. how can I test omniscidb ntile operation? normally we just compare the result from the backend to the result from pure pandas operation.

Reply:
I have no ideal what these operations actually do. Some documentation and examples would go along way. You need to write a much more explict test.

Reply:
sounds fair @jreback thanks for the feedback ... I will probably move ntile to a new PR and I will do it for sure. thanks!